src/trie.rs

```diff
@@ -273,16 +298,51 @@ where
             return Ok(());
         }
         let root = self.root.clone();
-        self.root = self.insert_at(root, Nibbles::from_raw(key, true), value.to_vec())?;
-        Ok(())
+        let path = &Nibbles::from_raw(key.clone(), true);
+        let result = self.insert_at(root, path, 0, value.to_vec());
+
+        if let Err(TrieError::MissingTrieNode {
+            node_hash,
+            traversed,
+            root_hash,
+            err_key: _,
+        }) = result
+        {
+            Err(TrieError::MissingTrieNode {
+                node_hash,
+                traversed,
+                root_hash,
+                err_key: Some(key),
+            })
+        } else {
+            self.root = result?;
```

Comment:
Is it possible this could be some other kind of error than `MissingTrieNode`? Then we wouldn't want to set that to root...